Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -192,6 +198,38 @@ struct CombinatorialKalmanFilterOptions {
   ///       certain surface
   const Surface* targetSurface = nullptr;
 
+  using BoundState = std::tuple<BoundTrackParameters, BoundMatrix, double>;
+
+  /// Delegate definition to create track states for selected measurements
+  /// @note expected to iterator over the given sourcelink range,
+  ///       select measurements, and create track states for
+  ///       which new tips are to be created, more over the outlier
+  ///       flag should be set for states that are outlier.
+  ///
+  /// @param geoContext The current geometry context
+  /// @param calibrationContext pointer to the current calibration context
+  /// @param surface the surface at which new track states are to be created
+  /// @param boundState the current bound state of the trajectory
+  /// @param slBegin Begin iterator for sourcelinks
+  /// @param slEnd End iterator for sourcelinks
+  /// @param prevTip Index pointing at previous trajectory state (i.e. tip)
+  /// @param buffer_trajectory a temporary trajectory which can be used to create temporary track states
+  /// @param trackStateCandidates a temporary buffer that can be used to collect track states
+  /// @param trajectory the trajectory to which the new states are to be added
+  /// @param logger a logger for messages
+  using StateCandidateCreator = Delegate<Result<boost::container::small_vector<
+      typename traj_t::TrackStateProxy::IndexType, s_maxBranchesPerSurface>>(
+      const Acts::GeometryContext& geoContext,
+      const Acts::CalibrationContext* calibrationContext,
```

Comment:
should this also be a reference?

Reply:
The calibrationContext ? I think it can be a nullptr.

Reply:
I think not having a reference wrapper in the actor is a convenience so we do not have to pass the reference at construction but can just reset the pointer whenever. The origin of the context should be a reference yes https://github.com/acts-project/acts/blob/36811859790b86bf7729bff2241d52ada0e32e27/Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp#L157

Reply:
I am bit confused now. In the original CKF (v35.0,0) The actor seems to store a pointer to the calibrationContext, this is passed as an argument to the calibrator, but the calibrator seems to expect a reference .... Not clear how this works. Likely it works because the calibration context is not used and then the pointer argument creates a temporary calibrationContext with the pointer to the original calibration context as payload. But maybe I overlook something.

Reply:
That sounds like a bug. `CalibrationContext` is a typedef to `std::any` so I guess it could convert from anything? Maybe it is worth to box the `std::any` and have an explicit constructor for `CalibrationContext`

Reply:
Would make sense to prevent such mistakes in the future. But this will break client code. Maybe for the time being the code could just be changed such that it is more explicit where calibrationContext means a pointer e.g.:
```
 @@ -317,7 +317,7 @@ class CombinatorialKalmanFilter {
     bool energyLoss = true;
 
     /// Calibration context for the finding run
-    const CalibrationContext* calibrationContext{nullptr};
+    const CalibrationContext* calibrationContextPtr{nullptr};
 
     /// @brief CombinatorialKalmanFilter actor operation
     ///
@@ -784,7 +784,7 @@ class CombinatorialKalmanFilter {
         ts.setReferenceSurface(boundParams.referenceSurface().getSharedPtr());
 
         // now calibrate the track state
-        m_extensions.calibrator(gctx, calibrationContext, sourceLink, ts);
+        m_extensions.calibrator(gctx, *calibrationContextPtr, sourceLink, ts);

```